rules/camunda-cloud/ad-hoc-sub-process.js

```diff
@@ -3,25 +3,53 @@ const { is, isAny } = require('bpmnlint-utils');
 const { reportErrors } = require('../utils/reporter');
 
 const { skipInNonExecutableProcess } = require('../utils/rule');
+const { greaterOrEqual } = require('../utils/version');
 
-module.exports = skipInNonExecutableProcess(function() {
+const COMPLETION_ATTRIBUTES_SUPPORT_VERSION = '8.8';
+
+module.exports = skipInNonExecutableProcess(function({ version }) {
   function check(node, reporter) {
     if (!is(node, 'bpmn:AdHocSubProcess')) {
       return;
     }
 
     // Ad-Hoc Sub-Process must contain at least one activity
-    if (node.get('flowElements').some(isActivity)) {
-      return;
+    if (!node.get('flowElements').some(isActivity)) {
+      reportErrors(node, reporter, {
+        message: 'Element of type <bpmn:AdHocSubProcess> must contain at least one activity',
+        data: {
+          node,
+          parentNode: null
+        }
+      });
     }
 
-    reportErrors(node, reporter, {
-      message: 'Element of type <bpmn:AdHocSubProcess> must contain at least one activity',
-      data: {
-        node,
-        parentNode: null
+    if (!greaterOrEqual(version, COMPLETION_ATTRIBUTES_SUPPORT_VERSION)) {
```

Comment:
Let's use the [`hasProperties` or `hasProperty` helper](https://github.com/camunda/bpmnlint-plugin-camunda-compat/blob/main/rules/utils/element.js#L185) for the checks. This decouples the error message from the rule, and also attaches the metadata (error type) consistently.

Reply:
Example: https://github.com/camunda/bpmnlint-plugin-camunda-compat/blob/9594d5036641bbbeb8835d3bdeae7bf02f70f016/rules/camunda-cloud/subscription.js#L52

Reply:
Another one with allowedVersion: https://github.com/camunda/bpmnlint-plugin-camunda-compat/blob/9594d5036641bbbeb8835d3bdeae7bf02f70f016/rules/camunda-cloud/no-template.js#L12